agenthub/micro/agent.py

```diff
@@ -44,6 +58,7 @@ def step(self, state: State) -> Action:
             state=state,
             instructions=instructions,
             to_json=to_json,
+            history_to_json=history_to_json,
```

Comment:
I think you're right we should save to_memory(). I just wonder, why wouldn't we just send here the json based on to_memory()?

Reply:
You mean just translate the history by calling `to_memory()` outside, and pass the result to `self.prompt_template.render`, so we don't need to call `history_to_json` method when rendering?   

If you mean this, I think this suggestion is ok  to me, what's your opinion @rbren , it seems you're the initial author.

Reply:
Perhaps even more, I think there's code in other agents that avoid to save "heavy stuff like screenshots for the user" to `history` in the first place.

Reply:
Yes, agree. This pr will just fix micro-agent `history` with heavy stuff like screenshots. I suggest to merge this, and later maybe we can refactor the Action/Observation history send to llm strategy, it should have unified processing method.